code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -189,6 +190,15 @@
 			M.IgniteMob()
 
 /mob/living/simple_animal/hostile/guardian/fire/Bumped(AM as mob|obj)
+	..()
+	if(istype(AM, /mob/living/))
+		var/mob/living/M = AM
+		if(AM != src.summoner)
+			M.adjust_fire_stacks(7)
+			M.IgniteMob()
```

Comment:
Three exact cvopies of the same code in all procs
Move it to separate proc and call it from the places you need